macosx/PrefsController.m

```diff
@@ -845,7 +845,8 @@ - (void) folderSheetShow: (id) sender
             [fDefaults setBool: YES forKey: @"DownloadLocationConstant"];
             [self updateShowAddMagnetWindowField];
             
-            tr_sessionSetDownloadDir(fHandle, [folder UTF8String]);
+            assert(folder.length > 0);
```

Comment:
Let's merge this in as is now and see what happens then (kidding, nothing bad expected) :) Thanks!